src/components/SignUp/index.js

```diff
@@ -1,19 +1,65 @@
-import React from 'react'
+import React, { useState } from 'react'
+import { Link } from 'react-router-dom'
+import { useMutation } from '@apollo/react-hooks'
+import gql from 'graphql-tag'
+
 import styles from './SignUp.module.css'
+import { Form, Button, Jumbotron, Container, Alert } from 'react-bootstrap'
 
-import Form from 'react-bootstrap/Form'
-import Button from 'react-bootstrap/Button'
-import Jumbotron from 'react-bootstrap/Jumbotron'
-import Container from 'react-bootstrap/Container'
-import { Link } from 'react-router-dom'
+const SIGN_UP = gql`
+  mutation sign_up($firstname:String!, $lastname:String!, $username: String!, $password: String!){
+    signup(data:{firstName:$firstname,lastName:$lastname,username:$username,password:$password}){
+      user{
+        id
+        username
+      }
+      jwt
+    }
+  }
+`
```

Comment:
You could've written this using the input type of the server. Example (assuming that `data` is of input type `SignupInput` on the server):
```suggestion
const SIGN_UP = gql`
  mutation sign_up($data: SignupInput!){
    signup(data: $data){
      user{
        id
        username
      }
      jwt
    }
  }
`
```

This way, you don't have to worry if new fields are added / existing ones are removed.